src/components/VtkThreeView.vue

```diff
@@ -323,16 +323,26 @@ export default defineComponent({
           light.setPositional(false);
         }
 
-        const sampleDistance =
-          1.5 *
-          Math.sqrt(
-            image
-              .getSpacing()
-              .map((v) => v * v)
-              .reduce((sum, v) => sum + v, 0)
-          );
+        mapper.setMaximumSamplesPerRay(1500);
+
+        const dims = image.getDimensions();
+        const spacing = image.getSpacing();
+        const spatialDiagonal = vec3.length(
+          vec3.fromValues(
+            dims[0] * spacing[0],
+            dims[1] * spacing[1],
+            dims[2] * spacing[2]
+          )
+        );
 
-        mapper.setSampleDistance(sampleDistance / 10);
+        let sampleDistance = (1.5 * vec3.length(spacing)) / 10;
+        if (
+          spatialDiagonal / sampleDistance >
+          mapper.getMaximumSamplesPerRay()
+        ) {
+          sampleDistance = spatialDiagonal / mapper.getMaximumSamplesPerRay();
```

Comment:
Perhaps divide by (mapper.getMaximumSamplesPerRay()+1); to avoid precision resulting in 1 too many samples.

Reply:
No more crashes for me on Chrome on Windows!

I do see the warning message persists:  macros.js:60 The number of steps required 1501 is larger than the
        specified maximum number of steps 1500.
        Please either change the
        volumeMapper sampleDistance or its maximum number of samples.

Reply:
I changed it to `maxSamplesPerRay-1`, which handles the off-by-one with the number of steps.